enhancements/sig-policy/66-delete-policy-resources/README.md

```diff
@@ -0,0 +1,224 @@
+the field be set at the template level. If users would like some objects to
+remain on deletion and others to be deleted, they can separate those objects
+into multiple configurationPolicies contained under the same parent policy.
+
+ConfigurationPolicies already maintain a list of objects they manage in their
+status field called `relatedObjects`. Since most of the information we would need
+to track which objects need to be deleted already exists in this part of the status,
+we propose adding a boolean field inside `status.relatedObjects.Object` that tells
+the controller whether a resource needs to be deleted when the configurationPolicy
+is deleted. We would set this value when the configurationPolicy controller creates
+a resource during the handling of an “enforce” policy template. When processing
+`spec.removeChildrenOnDelete`, we will set a finalizer on the configurationPolicy,
+such as `policy.open-cluster-management.io/delete-related-objects`. When the policy
+is deleted, the controller will perform cleanup logic on the objects in `relatedObjects`
+that have been marked for deletion by the new status field. Once that cleanup is complete,
+the finalizer will be removed and kubernetes will handle removing the configurationPolicy
+from the cluster. Any resources without this value will persist on the cluster.
+
+Since `removeChildrenOnDelete` and `spec.removeChildrenOnDelete` are scoped to the
+configurationPolicy level, the configurationPolicy controller will handle removing
+objects on each managed cluster once a configurationPolicy is deleted there. This also
+means that when the placement of a policy changes to a different managed cluster
+and that policy contains a configurationPolicy with `removeChildrenOnDelete: true`, the
+object will be deleted on the old cluster and created on the new one. This means that
+changing the placement of an `enforce` policy will no longer result in objects
+persisting indefinitely on the old managed cluster.
+
+
+### Risks and Mitigation
+
+The main risk of adding an enhancement like this would be users deleting objects
+that they want to remain on the cluster after they delete a policy. We plan to
+mitigate this risk by making the `spec.removeChildrenOnDelete` field default to
+`false`, so users will not delete anything without having to set that field first.
+
+## Design Details
+
+The `spec.removeChildrenOnDelete` field would look something like this:
+
+```
+spec:
+  namespaceSelector:
+    exclude:
+    - kube-*
+    include:
+    - default
+  object-templates:
+  - complianceType: musthave
+    objectDefinition:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: proposal-pod
+      spec:
+        containers:
+        - image: nginx:1.18.0
+          name: nginx
+          ports:
+          - containerPort: 80
+  remediationAction: enforce
+  severity: low
+  removeChildrenOnDelete: true //controller will remove proposal-pod
```

Comment:
will all objects in your templates list be deleted, or you want to optionally delete some.